src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -270,6 +266,23 @@ export class StudyViewPageStore {
        });
     }
 
+    // store reactions should be registered here.
+    onComponentDidMount() {
```

Comment:
this should really be in the same place where store is instantiated.   constructor of view. as  it actually has nothing to do with component mounting.

Reply:
you're right. I think it's my fault because I suggested it - I was confused and thought that the study view store was instantiated elsewhere, so that to put the call to destroy() in the same place as its instantiated would require something like this.